WoWsShipBuilder.ViewModels/Other/ShipComparisonViewModel.cs

```diff
@@ -0,0 +1,762 @@
+
+                    break;
+
+                case ShipComparisonDataSections.Bombs:
+                    if (!shipList.Any(x => x.BombersWeaponType.Any()))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+
+                case ShipComparisonDataSections.Sonar:
+                    if (!shipList.Any(x => x.ShipDataContainer.PingerGunDataContainer is not null))
+                    {
+                        dataSections.Remove(dataSection);
+                    }
+
+                    break;
+            }
+        }
+
+        return dataSections;
+    }
+
+    public async Task ToggleDataSection(ShipComparisonDataSections dataSection)
+    {
+        SelectedDataSection = dataSection;
+        await ApplyFilters();
+    }
+
+    public void SelectAllDisplayedShips()
+    {
+        SelectAllShips = !SelectAllShips;
+
+        List<ShipComparisonDataWrapper> list = GetShipsToBeDisplayedList();
+
+        if (!string.IsNullOrEmpty(searchString))
+        {
+            list = list.Where(x => appSettings.SelectedLanguage.CultureInfo.CompareInfo.IndexOf(localizer.GetGameLocalization(x.ShipIndex + "_FULL").Localization, searchString, CompareOptions.IgnoreNonSpace | CompareOptions.IgnoreCase) != -1).ToList();
+        }
+
+        if (SelectAllShips)
+        {
+            SelectedShipList.AddRange(list.Where(x => !ContainsWrapper(x, SelectedShipList)));
+        }
+        else
+        {
+            list.Where(x => ContainsWrapper(x, SelectedShipList)).ToList().ForEach(x => SelectedShipList.Remove(x));
+        }
+    }
+
+    public async Task PinAllDisplayedShips()
+    {
+        PinAllShips = !PinAllShips;
+
+        List<ShipComparisonDataWrapper> list = FilteredShipList;
+
+        if (!string.IsNullOrEmpty(searchString))
+        {
+            list = list.Where(x => appSettings.SelectedLanguage.CultureInfo.CompareInfo.IndexOf(localizer.GetGameLocalization(x.ShipIndex + "_FULL").Localization, searchString, CompareOptions.IgnoreNonSpace | CompareOptions.IgnoreCase) != -1).ToList();
+        }
+
+        if (PinAllShips)
+        {
+            PinnedShipList.AddRange(list.Where(x => !ContainsWrapper(x, PinnedShipList)));
+        }
+        else
+        {
+            list.Where(x => ContainsWrapper(x, PinnedShipList)).ToList().ForEach(x => PinnedShipList.Remove(x));
+            await ApplyFilters();
+        }
+    }
+
+    private void FindShips()
+    {
+        if (string.IsNullOrEmpty(searchShip.Trim()))
+        {
+            return;
+        }
+
+        SearchedShips.Clear();
+        SearchedShips.AddRange(fullShipList.Where(ship => appSettings.SelectedLanguage.CultureInfo.CompareInfo.IndexOf(localizer.GetGameLocalization(ship.Index + "_FULL").Localization, searchShip, CompareOptions.IgnoreNonSpace | CompareOptions.IgnoreCase) != -1));
+    }
+
+    public void DuplicateSelectedShips(ShipComparisonDataWrapper? ship = null)
```

Comment:
why does this get an optional parameter? Either use only the current state of the VM or only a parameter, don't combine two different cases into one